sympy/polys/numberfields.py

```diff
@@ -517,6 +517,8 @@ def _minpoly_compose(ex, x, dom):
     if ex is I:
         _, factors = factor_list(x**2 + 1, x, domain=dom)
         return x**2 + 1 if len(factors) == 1 else x - I
+    if ex is GoldenRatio:
+        return x**2 - x - 1
```

Comment:
Maybe this should be handled in the same way as `I`: It `GoldenRatio` is an element of `dom`, then `x**2 - x - 1` splits into two linear factors over `dom` and the minimal polynomial is one them.